cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/TransformExecutorFactory.java

```diff
@@ -131,7 +138,15 @@ public TransformExecutorFactory(JobContext hadoopContext, PipelinePluginInstanti
       return;
     }
 
-    addTransformation(pipeline, stageName, transformations, transformErrorSinkMap);
+    try {
+      addTransformation(pipeline, stageName, transformations, transformErrorSinkMap);
+    } catch (Exception e) {
+      // Catch the Exception to generate a User Error Log for the Pipeline
+      PIPELINELOG.error("Failed to start Pipeline Stage '{}' with error: {}. Please review your Pipeline" +
+                          " configuration and check system logs for more details.", stageName,
+                        Throwables.getRootCause(e).getMessage(), Throwables.getRootCause(e));
```

Comment:
Please change "with error:" to "with this error", and change "for more details" to "for details".